packages/go_router/lib/src/builder.dart

```diff
@@ -151,18 +104,19 @@ class GoRouterDelegate extends RouterDelegate<List<GoRouteMatch>>
 
     // pass either the match error or the build error along to the navigator
     // builder, preferring the match error
-    if (matches.length == 1 && matches.first.error != null) {
-      error = matches.first.error;
+    if (matches.matches.length == 1 && matches.matches.first.error != null) {
```

Comment:
maybe this can be a boolean property or a getter in RouteMatchList?

Reply:
Yeah that would be better - added.